game/src/devtools/polygon.rs

```diff
@@ -1,35 +1,37 @@
-use geom::{Circle, Distance, LonLat, Pt2D, Ring};
+use geom::{Circle, Distance, FindClosest, LonLat, Pt2D, Ring};
+use widgetry::mapspace::{ObjectID, World, WorldOutcome};
 use widgetry::{
-    Color, EventCtx, GfxCtx, HorizontalAlignment, Key, Line, Outcome, Panel, State, Text,
+    Cached, Color, EventCtx, GfxCtx, HorizontalAlignment, Key, Line, Outcome, Panel, State,
     VerticalAlignment, Widget,
 };
 
 use crate::app::App;
 use crate::app::Transition;
-use crate::common::CommonState;
-
-const POINT_RADIUS: Distance = Distance::const_meters(10.0);
-// Localized and internal, so don't put in ColorScheme.
-const POINT_COLOR: Color = Color::RED;
-const POLYGON_COLOR: Color = Color::BLUE.alpha(0.6);
-const POINT_TO_MOVE: Color = Color::CYAN;
-const LAST_PLACED_POINT: Color = Color::GREEN;
 
 pub struct PolygonEditor {
     panel: Panel,
     name: String,
-    points: Vec<LonLat>,
-    mouseover_pt: Option<usize>,
-    moving_pt: bool,
+    points: Vec<Pt2D>,
+    // The points change size as we zoom out, so rebuild based on cam_zoom
```

Comment:
This is a somewhat unusual extra requirement just present here. I thought about trying to wrap this kind of behavior up in the `World` as a configurable option, but I don't think it's worth that.